R/palaeorotate.R

```diff
@@ -218,12 +236,34 @@ palaeorotate <- function(occdf, lng = "lng", lat = "lat", age = "age",
     stop("`round` should be NULL or of class numeric")
   }
 
-  if (!is.logical(uncertainty)) {
-    stop("`uncertainty` should be a logical value: TRUE/FALSE")
+  if (!is.null(uncertainty) && !is.character(uncertainty)) {
+    stop("`uncertainty` should be NULL or a vector of model names")
+  }
+
+  # Add warnings for use of mantle reference frame models
+  if (any(model %in% c("MULLER2022", "MULLER2019",
+                   "MATTHEWS2016_mantle_ref"))) {
+    warning(paste0("Selected model(s) use a mantle reference frame and are ",
+    "not recommended for reconstructing palaeocoordinates. See details."))
+  }
+
+  if (any(uncertainty %in% c("MULLER2022", "MULLER2019",
```

Comment:
The first if statement is checking the `model` input argument and the second is checking the `uncertainty` input argument. This could be collapsed into a single if statement but I am not sure it makes too much of a difference.